pvlib/snowcoverage.py

```diff
@@ -0,0 +1,146 @@
+"""
+The ``snow`` module contains functions that model the effect of snow on
+solar modules.
+"""
+
+import numpy as np
+import pandas as pd
+from pvlib.tools import sind
+
+
+def _time_delta_in_hours(times):
+    delta = times.to_series().diff()
+    return delta.dt.seconds.div(3600)
+
+
+def fully_covered(snowfall, threshold=1.):
+    '''
+    Calculates the timesteps when the row's slant height is fully covered
+    by snow.
+
+    Parameters
+    ----------
+    snowfall : Series
+        Accumulated snowfall in each time period [cm]
+
+    threshold : float, default 1.0
+        Minimum hourly snowfall to cover a row's slant height. [cm/hr]
+
+    Returns
+    ----------
+    boolean: Series
+        True where the snowfall exceeds the defined threshold to fully cover
+        the panel.
+
+    Notes
+    -----
+    Implements the model described in [1]_ with minor improvements in [2]_.
+
+    References
+    ----------
+    .. [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+       “Measured and modeled photovoltaic system energy losses from snow for
+       Colorado and Wisconsin locations.” Solar Energy 97; pp.112-121.
+    .. [2] Ryberg, D; Freeman, J. "Integration, Validation, and Application
+       of a PV Snow Coverage Model in SAM" (2017) NREL Technical Report
+    '''
+    delta = snowfall.index.to_series().diff()  # [0] will be NaT
+    timestep = delta.dt.seconds.div(3600)  # convert to hours
+    time_adjusted = snowfall / timestep
+    time_adjusted.iloc[0] = 0  # replace NaN from NaT / timestep
+    return time_adjusted >= threshold
```

Comment:
I wonder if testing `time_adjusted > threshold` (`>` instead of `>=`) would be better.  Choosing `threshold=0` might be appropriate in some cases and using `>=` here would say zero snowfall is sufficient to fully cover the array.

Reply:
That's a good point. We'd need to rewrite the description of the `threshold_snowfall` parameter from 
```
    threshold_snowfall : float, default 1.0
        Minimum hourly snowfall to cover a row's slant height. [cm/hr]
```

to

```
    threshold_snowfall : float, default 1.0
        Hourly snowfall above which snow coverage is set to the row's slant height. [cm/hr]
```